library/common/jni/jni_interface.cc

```diff
@@ -60,11 +83,20 @@ static void jvm_on_exit(void*) {
 }
 
 extern "C" JNIEXPORT jint JNICALL Java_io_envoyproxy_envoymobile_engine_JniLibrary_runEngine(
-    JNIEnv* env, jclass, jlong engine, jstring config, jstring jvm_log_level, jobject context) {
-  jobject retained_context = env->NewGlobalRef(context); // Required to keep context in memory
-  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit, retained_context};
-  // TODO(junr03): wire up once Android support lands.
+    JNIEnv* env, jclass, jlong engine, jstring config, jstring jvm_log_level,
+    jobject on_start_context, jobject envoy_logger_context) {
+
+  // Creating a global reference to the on_start and envoy_logger contexts.
+  // This is required to keep context in memory.
+  jobject retained_on_start_context = env->NewGlobalRef(on_start_context);
+  jobject retained_logger_context = env->NewGlobalRef(envoy_logger_context);
+
+  envoy_engine_callbacks native_callbacks = {jvm_on_engine_running, jvm_on_exit,
+                                             retained_on_start_context};
   envoy_logger logger = {nullptr, nullptr, nullptr};
+  if (envoy_logger_context != nullptr) {
```

Comment:
I don't think I see how `envoy_logger_context` is optional? I actually am being silly and can't really even see how we go from the closure in kotlin to the Java EnvoyLogger interface? Is it magic?

Reply:
@buildbreaker did you verify that not using `setLogger` lets this be not set, and thus still see logs in logcat?

Reply:
> @junr03 Yep! It's going to logcat!

@buildbreaker awesome. Do you mind explaining how? I can't seem to understand how we go from the optional closure in kotlin to the jni, via a Java object that is (not?) optional?

Reply:
Everything in Java is a passed by reference so everything in Java is "optional". In this case, we can safely rely on the optionality of the `envoy_logger_context` to be our signal for if someone decided to register or not register a platform logger.